### ISSUE TITLE
Myrmex Queen AI

### ISSUE THREAD

**Issue by BorgerBoi** [Alex-the-666/Ice_and_Fire#4350]
So i've been trying to live inside a myrmex jungle hive, ive done it, but the queen got out and MADE A NEW hive, and reset my reputation back to 0, so ive been studying the myrmex queen, and it appears that her AI makes her leave the hive, create another hive try to get out again and build another hive, each time this happens the old hive's castes become rogue and attack anything

**Comment by PlayBonn** [comment]
I have found the same thing. In an old hive I use to use, the queen made a new hive and her old members killed her. Also as soon as she made it she started trying to leave the hive again (Entrance was to small for her to make out of so she just sat on the ceiling). I was able to trade for a queen egg before she wound up getting killed due to this bug. Queens that are born from eggs with you next to them show the same behavior, but when they make a new nest your reputation is dropped to 0. 

I wound up making a man made nest and used for chains in four corners of a small room to keep her in place. She winds up walking on the spot for a minute or two before launching forwards for a second and then snaps back into place. If it was just visual this wouldn't be a major game breaker but she tends to clip into walls while doing this and takes suffocation damage in spurts. I am lucky to have a mod that lets me heal other mobs, but I can't safely live in or look after this colony because the queen may wind up killing herself due to this.

**Comment by DigitalWendigo** [comment]
I've had several issues with the myrmex queen and her AI. 
Currently the only ways to somewhat battle issues like this is to ensure the exit and entrance of the hive, is too small for the queen to exit. 
She'll still try, but will be unable to exit because she can't fit through the gap. 
In short.. If you want to keep the colony alive and functioning, you'll need to modify the hive structure by hand. 
This will also potentially fix some issues with the other myrmex pathfinding, as the workers tend to just some times stand in place, unable to deposit items in the storage room. 
This also means you need to first delete all the already existing rooms, and re-add them after you patched holes.

**Comment by DigitalWendigo** [comment]
Please update to the latest Alpha release found here https://github.com/Alex-the-666/Ice_and_Fire/releases/tag/2.1.10-alpha2 
This issue is no longer present in the last alpha. 
This can be closed.

**Comment by TheViralDivinity** [comment]
> Please update to the latest Alpha release found here https://github.com/Alex-the-666/Ice_and_Fire/releases/tag/2.1.10-alpha2 This issue is no longer present in the last alpha. This can be closed.

Incorrect. Having this issue in 2025, in 1.20.1, and multiple people have been since 2023

**Comment by ciekals11** [comment]
Adding to above. Myrmexes (not only queen) appears to be teleporting instead of walking if path is obstructed so if you live outside of colony they may tp to you and attack you \
screen of that happening with queen (colony is behind wall visible behind her) \
![Image](https://github.com/user-attachments/assets/a5164a61-5fc5-4658-97c2-2f9609ee3ce2) \
and related issue. Queen is so big that after she teleports, she becomes stuck in walls and suffocate.

Keeping her alive is nightmare
